### PR TITLE
Feature/electricity service

### PR DESCRIPTION
**Pull request by Daan298261**
full commit

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,261 @@
+## Ignore Visual Studio temporary files, build results, and
```
Comment (markieo1) [inline .gitignore:1]: Please update this file to the latest version on [VisualStudio.gitignore](https://github.com/github/gitignore/blob/master/VisualStudio.gitignore)

File: ElectricityService.App/Properties/launchSettings.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (markieo1) [inline ElectricityService.App/Properties/launchSettings.json]: File should not exist in the repository, can be fixed by updating the gitignore

File: ElectricityService.Core/Models/Ship.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+{
+   public class Ship
+    {
+
+        [Key]
+        public string Serial { get; set; }
+        public string Name { get; set; }
+
+        [IgnoreDataMember]
+        public ShipService ShipService { get; set; }
```
Comment (markieo1) [inline ElectricityService.Core/Models/Ship.cs]: This should not be on a ship?

File: ElectricityService.Infrastructure/Services/ElectricityService.cs [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+        public Task SendServiceCompletedAsync(Ship ship)
+        {
+            return Task.Run(async () =>
+            {
+              
+                ShipService shipService = new ShipService();
+                shipService.Name = "Electricity";
+                shipService.Id = new Guid();
+                shipService.Price = 2500;
```
Comment (markieo1) [inline ElectricityService.Infrastructure/Services/ElectricityService.cs]: The electricity service does not decide the price!

File: README.md [hunk 5]
```diff
@@ -1 +0,0 @@
-# ElectricityService
```
Comment (markieo1) [inline README.md]: No need for this change.

**Review by markieo1 — DISMISSED** [review]

File: ElectricityService.Core/Models/Ship.cs (hunk 3 above)
Comment (Daan298261) [reply ElectricityService.Core/Models/Ship.cs]: you said 'look in my project for it' so i did.

File: ElectricityService.Infrastructure/Services/ElectricityService.cs (hunk 4 above)
Comment (Daan298261) [reply ElectricityService.Infrastructure/Services/ElectricityService.cs]: according to the data that kevin requires it does

File: ElectricityService.App/Properties/launchSettings.json (hunk 2 above)
Comment (Daan298261) [reply ElectricityService.App/Properties/launchSettings.json]: did so, @Aplicticus can verify Still VS insists that the file should be committed...

**Review by Aplicticus — APPROVED** [review]

File: ElectricityService.Infrastructure/DI/DIHelper.cs [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    {
+        public static void Setup(IServiceCollection services, IConfiguration configuration)
+        {
+
+            services.AddDbContext<ElectricityDbContext>(options => options.UseSqlServer(configuration.GetSection("DB_CONNECTION_STRING").Value));
+
+            services.AddTransient<IShipRepository, ShipRepository>();
+            services.AddTransient<IElectricityService, Services.ElectricityService>();
+
+            services.AddSingleton<IEventHandler, RMQMessageHandler>((provider) => new RMQMessageHandler(configuration.GetSection("AMQP").Value));
```
Comment (markieo1) [inline ElectricityService.Infrastructure/DI/DIHelper.cs]: This causes issues on production see [App doesnt start in production](https://github.com/GiethoornInternationalHarbor/RefuelService/issues/2)

**Review by markieo1 — CHANGES_REQUESTED** [review]

File: ElectricityService.Infrastructure/DI/DIHelper.cs (hunk 6 above)
Comment (Daan298261) [reply ElectricityService.Infrastructure/DI/DIHelper.cs]: has been fixed in new branch called hotfix/AMQP_URL. i am closing this PR and opening a new one on that branch. it should launch without any issues now.

**Comment by Daan298261** [comment]
Migrated to branch hotfix/AMQP_URL

**Comment by markieo1** [comment]
Please merge that branch in to this one. Then we do not lose alll the comments ;-)

**Comment by Daan298261** [comment]
Merged hotfix into this as per your request

**Review by markieo1 — APPROVED** [review]

**Pull request merged**